src/ios/VLCPlugin.m

```diff
@@ -1014,6 +1138,117 @@ - (void)vlc_checkForAndCreateLibraryNoCloudDirectory {
+    [self stopPlayback];
+}
+
+- (void)audioPlayerSetAudioSaved:(NYPRAudio *)audio saved:(BOOL)saved {
+    [self setSavedAudio:audio];
+}
+
+- (void)audioPlayerGetAudioSaved:(NYPRAudio *)audio {
+    [self getSavedStatusForAudio:audio];
+}
+
+#pragma mark NYPRAudioPlayerViewControllerDelegate methods
+
+- (void)nextButtonTapped {
+    if (self.isDiscoverAudioSelected) {
+        [self playNextDiscoverTrack];
+    } else if (self.isPlayingSavedAudioQueue) {
+        [self initiateNextSavedQueueAudioPlayback];
+    }
+}
+
+- (void)previousButtonTapped {
+    if ([[self.mediaPlayer time] intValue] > kVLCPluginRewindThreshold && [self.audio isKindOfClass:[NYPROnDemand class]]) {
+        [self seekToPosition:0];
+        return;
+    }
+    
+    if (self.isDiscoverAudioSelected) {
+        [self.discoverAudioPlaybackDelegate discoverPlayerPreviousTrack];
+    } else if (self.isPlayingSavedAudioQueue) {
+        [self initiatePreviousSavedQueueAudioPlayback];
+    }
+}
+
+- (void)rewindButtonTapped:(NSInteger)interval {
+    [self seekInterval:interval withCallbackId:self.callbackId];
+}
+
+- (void)playButtonTapped {
+    if (self.mediaPlayer.state != VLCMediaPlayerStatePlaying) {
+        [self playAudio:self.audio];
+    }
+}
+
+- (void)pauseButtonTapped {
+    if (self.mediaPlayer.state != VLCMediaPlayerStatePaused) {
+        [self.mediaPlayer pause];
+    }
+}
+
+- (void)seekToPosition:(NSInteger)position {
+    [self seekTo:position withCallbackId:self.callbackId];
+}
+
+- (void)visualPlayerSetAudioSaved:(NYPRAudio *)audio saved:(BOOL)saved {
+    [self setSavedAudio:audio];
+}
+
+- (void)visualPlayerGetAudioSaved:(NYPRAudio *)audio {
+    [self getSavedStatusForAudio:audio];
+}
+
+#pragma mark Saved Audio Methods
+
+- (void)setSavedAudio:(NYPRAudio *)audio {
+    NSDictionary *pluginResultMessage = @{ kVLCPluginJSONTypeKey : kVLCPluginSaveAudioTypeKey,
+                                           kVLCPluginSaveAudioTypeKeyAudio : [audio metadataAsDictionary]};
+    
+    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:pluginResultMessage];
+    [self vlc_sendPluginResult:pluginResult callbackId:self.callbackId];
+    [self.audioPlayerViewController visualPlayerSetAudioSaved:audio saved:YES];
+}
+
+- (void)getSavedStatusForAudio:(NYPRAudio *)audio {
+    NSDictionary *pluginResultMessage = @{ kVLCPluginJSONTypeKey : kVLCPluginIsAudioSaveAudioTypeKey,
+                                           kVLCPluginSaveAudioTypeKeyAudio : [audio metadataAsDictionary]};
+    
+    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:pluginResultMessage];
+    [self vlc_sendPluginResult:pluginResult callbackId:self.callbackId];
+}
+
+- (void)audioSavedStatus:(CDVInvokedUrlCommand*)command {
+    BOOL isSaved = [command.arguments.lastObject boolValue];
+    [self.audioPlayerViewController visualPlayerSetAudioSaved:self.audio saved:isSaved];
```

Comment:
calls should be:
 [self.audioPlayerViewController visualPlayerSetAudioSaved:audio saved:isSaved];
[self.discoverAudioPlaybackDelegate audioPlayerSetAudioSaved:audio saved:isSaved];

where audio is the audio object whose status was requested. this might not be the currently playing audio -- it could be audio 'saved' from a Discover story detail screen navigated to via the Discover playlist.